chunky/src/java/se/llbit/chunky/renderer/scene/Scene.java

```diff
@@ -2358,45 +2316,14 @@ public void setBufferFinalization(boolean value) {
    * Merge a render dump into this scene.
    */
   public void mergeDump(File dumpFile, TaskTracker taskTracker) {
-    int dumpSpp;
-    long dumpTime;
-    try (TaskTracker.Task task = taskTracker.task("Merging render dump", 2);
-        DataInputStream in = new DataInputStream(
-            new GZIPInputStream(new FileInputStream(dumpFile)))) {
-      task.update(1);
-      Log.info("Loading render dump " + dumpFile.getAbsolutePath());
-      int dumpWidth = in.readInt();
-      int dumpHeight = in.readInt();
-      if (dumpWidth != width || dumpHeight != height) {
-        Log.warn("Render dump discarded: incorrect width or height!");
-        return;
-      }
-      dumpSpp = in.readInt();
-      dumpTime = in.readLong();
-
-      double sa = spp / (double) (spp + dumpSpp);
-      double sb = 1 - sa;
-
-      for (int x = 0; x < width; ++x) {
-        task.update(width, x + 1);
-        for (int y = 0; y < height; ++y) {
-          samples[(y * width + x) * 3 + 0] =
-              samples[(y * width + x) * 3 + 0] * sa + in.readDouble() * sb;
-          samples[(y * width + x) * 3 + 1] =
-              samples[(y * width + x) * 3 + 1] * sa + in.readDouble() * sb;
-          samples[(y * width + x) * 3 + 2] =
-              samples[(y * width + x) * 3 + 2] * sa + in.readDouble() * sb;
-          finalizePixel(x, y);
-        }
-      }
-      Log.info("Render dump loaded");
-
-      // Update render status.
-      spp += dumpSpp;
-      renderTime += dumpTime;
+    Log.info("Merging render dump: " + dumpFile);
+    try {
+      RenderDump.merge(dumpFile, this, taskTracker);
+      postProcessFrame(taskTracker);
     } catch (IOException e) {
-      Log.info("Render dump not loaded");
+      Log.warn("IO exception while merging render dump!", e);
     }
+    Log.info("Render dump merged: " + dumpFile);
```

Comment:
This should be moved before line 2323. Otherwise it will be printed even if merging failed.